### ISSUE TITLE
[Bug]: checkpoint replay memory leak

### ISSUE THREAD

**Issue by XuPeng-SH** [matrixorigin/matrixone#14203]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main,1.1-dev

### Commit ID

7d9ca8fbd14c3f8019e15661acaaa6e2570d5d70

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

 mpool leak after replaying checkpoint

### Steps to Reproduce

```Markdown
1. Create some checkpoint
2. Restart
```

**Comment by XuPeng-SH** [comment]
assigned to @gouhongshen

**Comment by XuPeng-SH** [comment]
fixed by #14221
